WooCommerce/Classes/ViewRelated/Orders/Order Creation/OrderForm.swift

```diff
@@ -297,7 +301,23 @@ private struct ProductsSection: View {
                     }
                 })
                 Button(OrderForm.Localization.addProductViaSKUScanner) {
-                    showAddProductViaSKUScanner.toggle()
+                    let capturePermissionStatus = viewModel.capturePermissionStatus
+                    switch capturePermissionStatus {
+                    case .notPermitted:
+                        logPermissionStatus(status: .notPermitted)
+                        self.showPermissionsSheet = true
+                    case .notDetermined:
+                        logPermissionStatus(status: .notDetermined)
+                        viewModel.requestCameraAccess(onCompletion: { isPermissionGranted in
+                            if isPermissionGranted {
+                                showAddProductViaSKUScanner.toggle()
+                                logPermissionStatus(status: .permitted)
+                            }
+                        })
+                    case .permitted:
+                        showAddProductViaSKUScanner.toggle()
```

Comment:
Should these `toggle` calls just be `= true`? I don't think we should ever be changing them to `false` when we tap the button, so being more specific here might be clearer.

Reply:
You're right! Updated: https://github.com/woocommerce/woocommerce-ios/pull/9687/commits/0fcc516f296adf1ab57bfbaa9599d34050a929c5